### PR TITLE
[PICA-76] wrap LockConfig::duration_presets in an enum

### PR DESCRIPTION
**Pull request by benluelo**
https://app.clickup.com/t/20465559/PICA-76

### CODE REVIEW

File: code/parachain/frame/composable-traits/src/staking/lock.rs [hunk 1]
```diff
@@ -16,13 +16,52 @@ use crate::time::{DurationSeconds, Timestamp};
...
 }
 
+/// Represents both valid staking lock durations and ther associated multiplier bonus.
+#[derive(
+	DebugNoBound, PartialEqNoBound, EqNoBound, CloneNoBound, MaxEncodedLen, Encode, Decode, TypeInfo,
+)]
+#[scale_info(skip_type_params(MaxDurationPresets))]
+pub enum DurationMultipliers<MaxDurationPresets: Get<u32>> {
+	/// Fixed duration multipliers mapped to their respective multipliers.
+	// TODO(benluelo): Wrap this in `Validated` to ensure that at least one preset is provided?
```
Comment (vimukthi-git) [inline code/parachain/frame/composable-traits/src/staking/lock.rs:32]: We can avoid TODO for this types of comments, it's confusing to readers. Let's just make it "Possible Improvement" or similar

Comment (vimukthi-git) [reply code/parachain/frame/composable-traits/src/staking/lock.rs:32]: Besides that, we can also have a default for a pool or do validation both are fine by me

File: code/parachain/frame/composable-traits/src/staking/lock.rs [hunk 2]
```diff
@@ -16,13 +16,52 @@ use crate::time::{DurationSeconds, Timestamp};
...
 	/// The possible locking duration.
-	pub duration_presets:
-		BoundedBTreeMap<DurationSeconds, Validated<FixedU64, GeOne>, MaxDurationPresets>,
+	pub duration_multipliers: DurationMultipliers<MaxDurationPresets>,
 	/// The penalty applied if a staker unstake before the end date.
 	/// In case of zero penalty, you cannot unlock before it duration ends.
 	pub unlock_penalty: Perbill,
 }
 
+/// Represents both valid staking lock durations and ther associated multiplier bonus.
```
Comment (vimukthi-git) [inline code/parachain/frame/composable-traits/src/staking/lock.rs:25]:
```suggestion
/// Represents both valid staking lock durations and ther associated share multiplier bonus.
```

File: code/parachain/frame/composable-traits/src/staking/lock.rs [hunk 3]
```diff
@@ -16,13 +16,52 @@ use crate::time::{DurationSeconds, Timestamp};
 #[scale_info(skip_type_params(MaxDurationPresets))]
```
Comment (vimukthi-git) [inline code/parachain/frame/composable-traits/src/staking/lock.rs:16]: I prefer if we just remove this lock.rs file entirely and keep everything related to staking in a single file

**Review by vimukthi-git — APPROVED** [review]
LGTM

File: code/parachain/frame/composable-traits/src/staking/lock.rs (hunk 1 above)
Comment (benluelo) [reply code/parachain/frame/composable-traits/src/staking/lock.rs:32]: I don't think it's confusing for readers; this TODO will be addressed in a followup PR anyways. I wanted to keep this PR as simple as possible, not introducing any changes other than wrapping the presets in the enum

**Comment by vimukthi-git** [comment]
merge conflict

**Review by PoisonPhang — APPROVED** [review]

**Review by PoisonPhang — APPROVED** [review]

**Review by PoisonPhang — APPROVED** [review]

File: code/parachain/frame/composable-traits/src/staking/lock.rs (hunk 1 above)
Comment (vimukthi-git) [reply code/parachain/frame/composable-traits/src/staking/lock.rs:32]: Work generally should not get increased with TODOs as that is unpredictable to understand when something is actually "done" and ready to release. We can create clickup tasks and prioritize if needed.

**Comment by benluelo** [comment]
closing in favor of #2999

**Pull request closed without merging**